src/Features/Core/Portable/EmbeddedLanguages/AbstractLanguageDetector.cs

```diff
@@ -12,9 +12,8 @@
 
 namespace Microsoft.CodeAnalysis.Features.EmbeddedLanguages
 {
-    internal abstract class AbstractLanguageDetector<TOptions, TTree>
+    internal abstract class AbstractLanguageDetector<TOptions>
```

Comment:
split this API into two parts for cleanliness.  the base type doesn't care about building trees, the derived contains only teh logic for that.